lib/OpenQA/CLI/schedule.pm

```diff
@@ -0,0 +1,97 @@
+
+package OpenQA::CLI::schedule;
+use OpenQA::Jobs::Constants;
+use Mojo::Base 'OpenQA::Command', -signatures;
+use Mojo::Util qw(getopt);
+use Term::ANSIColor qw(colored);
+use open qw(:std :encoding(UTF-8));
+
+has description => 'Schedules a set of jobs (via "isos post" creating a schedule product)';
+has usage => sub { shift->extract_usage };
+has post_url => sub { shift->url_for('isos') };
+
+sub _create_jobs ($self, $client, $args, $param_file, $job_ids) {
+    my $params = $self->parse_params($args, $param_file);
+    my $tx = $client->start($client->build_tx(POST => $self->post_url, {}, form => $params));
+    my $res = $self->handle_result($tx, {pretty => 0, quiet => 1, links => 0, verbose => 0});
+    return $res if $res != 0;
+    my $json = $tx->res->json;
+    push @$job_ids, $json->{id} if defined $json->{id} && ref $json->{id} eq '';
+    push @$job_ids, @{$json->{ids}} if ref $json->{ids} eq 'ARRAY';
+    return 0 unless my $error = $json->{error};
+    print STDERR colored(['red'], $error, "\n");
+    return 1;
+}
+
+sub _monitor_jobs ($self, $client, $poll_interval, $job_ids, $job_results) {
+    while (@$job_results < @$job_ids) {
+        my $job_id = $job_ids->[@$job_results];
+        my $tx = $client->start($client->build_tx(GET => $self->url_for("experimental/jobs/$job_id/status"), {}));
+        my $res = $self->handle_result($tx, {pretty => 0, quiet => 1, links => 0, verbose => 0});
+        return $res if $res != 0;
+        my $job = $tx->res->json;
+        my $job_state = $job->{state} // NONE;
+        if (OpenQA::Jobs::Constants::meta_state($job_state) eq OpenQA::Jobs::Constants::FINAL) {
+            push @$job_results, $job->{result} // NONE;
+            next;
+        }
+        print "Job state of job ID $job_id: $job_state, waiting…\n";
+        sleep $poll_interval;
+    }
+}
+
+sub _compute_return_code ($self, $job_results) {
+    for my $job_result (@$job_results) {
+        return 2 unless OpenQA::Jobs::Constants::is_ok_result($job_result);
+    }
+    return 0;
+}
+
+sub command ($self, @args) {
+    die $self->usage
+      unless getopt \@args,
+      'param-file=s' => \my @param_file,
+      'm|monitor' => \my $monitor,
+      'i|poll-interval=i' => \my $poll_interval,
+      'r|set-return-code' => \my $set_return_code;
+    @args = $self->decode_args(@args);
+    my $client = $self->client($self->post_url);
+
+    my @job_ids;
+    my $create_res = $self->_create_jobs($client, \@args, \@param_file, \@job_ids);
+    return $create_res if $create_res != 0 || !($monitor || $set_return_code);
+
+    my @job_results;
+    my $monitor_res = $self->_monitor_jobs($client, $poll_interval // 10, \@job_ids, \@job_results);
+    return $monitor_res if $monitor_res != 0 || !$set_return_code;
+    return $self->_compute_return_code(\@job_results);
+}
+
+1;
+
+=encoding utf8
+
+=head1 SYNOPSIS
+
+  Usage: openqa-cli schedule [OPTIONS] DISTRI=… VERSION=… FLAVOR=… ARCH=… [ISO=… …]
```

Comment:
does it still work to specify parameters in lower-case? Then I would prefer that form or at least mention it as a convenience option.

Reply:
We could have a flag that auto-uppercases all parameters.

Reply:
ok, nevermind for now. It's something unrelated to support lowercase arguments automatically turned into uppercase. You are welcome to discuss my proposal in https://github.com/os-autoinst/openQA/pull/5038